File: backend/src/main/java/com/festago/dto/MemberTicketsResponse.java
package com.festago.dto;

import static java.util.stream.Collectors.collectingAndThen;
import static java.util.stream.Collectors.toList;

import com.festago.domain.MemberTicket;
import java.util.List;

public record MemberTicketsResponse(List<MemberTicketResponse> tickets) {

    public static MemberTicketsResponse from(List<MemberTicket> memberTickets) {
        return memberTickets.stream()
                .map(MemberTicketResponse::from)
                .collect(collectingAndThen(toList(), MemberTicketsResponse::new));
    }
}

File: android/festago/app/src/main/java/com/festago/festago/data/dto/MemberTicketsResponse.kt
package com.festago.festago.data.dto

import com.festago.festago.domain.model.Ticket
import kotlinx.serialization.Serializable

@Serializable
data class MemberTicketsResponse(
    val memberTickets: List<MemberTicketResponse>,
) {
    fun toDomain(): List<Ticket> = memberTickets.map { it.toDomain() }
}
